service/src/test/java/greencity/service/HabitServiceImplTest.java

```diff
@@ -1250,4 +1256,33 @@ void checkAccessOfOwnerToCustomHabitThrowsUserHasNoPermissionToAccessExceptionTe
         verify(userRepo).findByEmail(user.getEmail());
         verify(habitRepo, times(0)).save(any(Habit.class));
     }
+
+    @Test
+    void likeTest() {
```

Comment:
please add verify to your test

Reply:
Please add the tests when
`habitRepo.findById(habitId)` and  `userRepo.findById(habit.getUserId())` throw exception

Reply:
And please add a text that we do not send a notification if `habitAuthor == null`